File: adapter/src/main/java/com/ashlikun/adapter/recyclerview/callback/ItemTouchHelperAdapter.java
package com.ashlikun.adapter.recyclerview.callback;

import android.graphics.Canvas;
import androidx.recyclerview.widget.RecyclerView;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/10/19　9:12
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：ItemTouchHelper 的回调再次封装的接口
 */
public interface ItemTouchHelperAdapter {
    public void onItemDragStart(RecyclerView.ViewHolder viewHolder);

    public void onItemSwipeStart(RecyclerView.ViewHolder viewHolder);

    public void onItemDragEnd(RecyclerView.ViewHolder viewHolder);

    public void onItemSwipeClear(RecyclerView.ViewHolder viewHolder);

    public boolean onItemDragMoving(RecyclerView.ViewHolder source, RecyclerView.ViewHolder target);

    public void onItemSwiped(RecyclerView.ViewHolder viewHolder);

    public void onItemSwiping(Canvas canvas, RecyclerView.ViewHolder viewHolder, float dX, float dY, boolean isCurrentlyActive);

    public boolean isItemSwipeEnable();

    /**
     * 这个holder是不是内部自己创建的，
     *
     * @return true 没有任何触摸事件
     */
    public boolean isViewCreateByAdapter(RecyclerView.ViewHolder viewHolder);
}

File: adapter/src/main/java/com/ashlikun/adapter/recyclerview/callback/ItemTouchHelperAdapter.kt
package com.ashlikun.adapter.recyclerview.callback

import android.graphics.Canvas
import androidx.recyclerview.widget.RecyclerView

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.3 15:50
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：ItemTouchHelper 的回调再次封装的接口
 */

interface ItemTouchHelperAdapter {
    var isItemSwipeEnable: Boolean
    var isItemDraggable: Boolean
    fun onItemDragStart(viewHolder: RecyclerView.ViewHolder)
    fun onItemSwipeStart(viewHolder: RecyclerView.ViewHolder)
    fun onItemDragEnd(viewHolder: RecyclerView.ViewHolder)
    fun onItemSwipeClear(viewHolder: RecyclerView.ViewHolder)
    fun onItemDragMoving(source: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean
    fun onItemSwiped(viewHolder: RecyclerView.ViewHolder)
    fun onItemSwiping(canvas: Canvas, viewHolder: RecyclerView.ViewHolder, dX: Float, dY: Float, isCurrentlyActive: Boolean)

    fun isViewCreateByAdapter(viewHolder: RecyclerView.ViewHolder): Boolean
}